### ISSUE TITLE
CSS pseudo selector :nth-child is not translated to XPath correctly

### ISSUE THREAD

**Issue by mndl** [jOOQ/jOOX#158]
Hi, I am converting the css selector obtained from Jsoup to xpath using jOOX. I found that sometimes the conversion from nth-child() to position() is not correct.

### Expected behavior and actual behavior:

Actual Output:

**position() = 3**
```
Node name : i
      Tag : i
      CSS : #mc_embed_signup > p.intern:nth-child(3) > i
    XPath : //*[@id='mc_embed_signup']/p[@class='intern' or starts-with(@class, 'intern ') or ' intern' = substring(@class, string-length(@class) - string-length(' intern') + 1) or contains(@class, ' intern ')][position() = 3]/i
```

Expected Output:

**position() = 2**
```
Node name : i
      Tag : i
      CSS : #mc_embed_signup > p.intern:nth-child(3) > i
    XPath : //*[@id='mc_embed_signup']/p[@class='intern' or starts-with(@class, 'intern ') or ' intern' = substring(@class, string-length(@class) - string-length(' intern') + 1) or contains(@class, ' intern ')][position() = 2]/i
```

### Steps to reproduce the problem:
```
Document doc = Jsoup.connect("https://theuserisdrunk.com/").get();
Elements elements = doc.select("*");
  
  for (Element element : elements) {
    String path = CSS2XPath.css2xpath(element.cssSelector(), false);
    System.out.println("Node name : " + element.nodeName());
    System.out.println("      Tag : " + element.tagName());
    System.out.println("      CSS : " + element.cssSelector());
    System.out.println("    XPath : " + path + "\n");
}
```
### Versions:

- jOOX: 1.5.0
- jsoup: 1.10.3
- Java: Java 8

**Comment by lukaseder** [comment]
Thank you very much for reporting this. Indeed, it doesn't look like the current implementation is correct at all. `:nth-child(n)` should be translated to `[count(preceding-sibling::*) = n - 1]`, not to `[position() = n]`

**Comment by lukaseder** [comment]
Fixed for 1.6.1. Would you mind checking if the fix works for you?

**Comment by mndl** [comment]
@lukaseder  Perfect !!! Thank You.
